File: CS/FieldListDragDropService.cs
using System;
using System.ComponentModel.Design;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraReports.Design;
using DevExpress.XtraReports.UI;

namespace FieldListDragDrop {
    public class MyFieldListDragDropService : IFieldListDragDropService {
        IDesignerHost host;
        public MyFieldListDragDropService(IDesignerHost host) { this.host = host; }
        public IDragHandler GetDragHandler() {
            return new MyFieldDragHandler(host);
        }
    }
    public class MyFieldDragHandler : FieldDragHandler {
        IDesignerHost mhost;
        public MyFieldDragHandler(IDesignerHost host)
            : base(host) {
            this.mhost = host;
        }
        static bool MIsContolContainer(XRControl control) {
            return control is Band || control is XRPanel;
        }
        public override void HandleDragDrop(object sender, DragEventArgs e) {
            AdornerService.ResetSnapping();
            RulerService.HideShadows();

            XRControl parent = bandViewSvc.GetControlByScreenPoint(new Point(e.X, e.Y));
            if (parent == null)
                return;

            XRControl control = (XRControl)Activator.CreateInstance(typeof(XRControl));

            PointF basePoint = EvalBasePoint(e);
            RectangleF controlRect = new RectangleF(basePoint, control.BoundsF.Size);
            PointF dropPoint = bandViewSvc.PointToClientRelativeToBand(basePoint,
                new DragDataObject(new XRControl[] { control }, control, 
                    new RectangleF[] { controlRect }, controlRect));
            XRRichText xRLabel = new XRRichText();

            DesignTool.AddToContainer(host, xRLabel);

            PointF location = PointToClient(xRLabel, dropPoint);

            xRLabel.LocationF = location;
            xRLabel.Size = new Size(100, 25);
            xRLabel.DataBindings.Add("Rtf", null, "test");


        }
        static DragDataObject GetDragData(IDataObject dataObject) {
            return (DragDataObject)dataObject.GetData(typeof(DragDataObject));
        }
        PointF PointToClient(XRControl control, PointF screenPoint) {
            return ZoomService.GetInstance(host).
                FromScaledPixels(bandViewSvc.PointToClient(screenPoint, null, control), ReportDpi);
        }
        protected float ReportDpi {
            get {
                return ((XtraReport)host.RootComponent).Dpi;
            }
        }
    }
}

File: VB/FieldListDragDropService.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel.Design
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Data.Browsing
Imports DevExpress.XtraReports.Design
Imports DevExpress.XtraReports.UI

Namespace FieldListDragDrop
	Public Class MyFieldListDragDropService
		Implements IFieldListDragDropService
		Private host As IDesignerHost
		Public Sub New(ByVal host As IDesignerHost)
			Me.host = host
		End Sub
        Public Function GetDragHandler() As IDragHandler Implements IFieldListDragDropService.GetDragHandler
            Return New MyFieldDragHandler(host)
        End Function
	End Class
	Public Class MyFieldDragHandler
		Inherits FieldDragHandler
		Private mhost As IDesignerHost
		Public Sub New(ByVal host As IDesignerHost)
			MyBase.New(host)
			Me.mhost = host
		End Sub
		Private Shared Function MIsContolContainer(ByVal control As XRControl) As Boolean
			Return TypeOf control Is Band OrElse TypeOf control Is XRPanel
		End Function
		Public Overrides Sub HandleDragDrop(ByVal sender As Object, ByVal e As DragEventArgs)

            Dim dropPoint As System.Drawing.Point = New System.Drawing.Point(e.X, e.Y)
			Dim droppedData() As DataInfo = TryCast(e.Data.GetData(GetType(DataInfo())), DataInfo())
			Dim control As XRControl = bandViewSvc.GetControlByScreenPoint(dropPoint)

			If MIsContolContainer(control) Then
				Dim size As New Size(100, 25)
				Dim dragData As DragDataObject = GetDragData(e.Data)
				Dim location As PointF = bandViewSvc.PointToClient(dropPoint, dragData, control)
				Dim l As XRRichText = Nothing
				For i As Integer = 0 To droppedData.Length - 1
					l = New XRRichText()
					l.Size = size
					l.Location = System.Drawing.Point.Ceiling(location)
					l.DataBindings.Add("Rtf", droppedData(i).Source, droppedData(i).Member)
					ReportDesigner.AddToContainer(mhost, l)
					control.Controls.Add(l)
					location.X += 100
				Next i
			Else
				MyBase.HandleDragDrop(sender, e)
			End If

		End Sub

		Private Shared Function GetDragData(ByVal dataObject As IDataObject) As DragDataObject
			Return CType(dataObject.GetData(GetType(DragDataObject)), DragDataObject)
		End Function

	End Class
End Namespace